interface/views.py

```diff
@@ -71,7 +71,12 @@ def upload(request, course_code, assignment_code):
         form = UploadFileForm(request.POST, request.FILES)
         if form.is_valid():
             file = request.FILES['file']
-            handle_submission(file, assignment, request.user)
+            delta_t = handle_submission(file, assignment, request.user)
+
+            if delta_t:
+                return render(request, 'interface/upload.html',
+                              {'form': form, 'delta_t': delta_t})
```

Comment:
I'm super biased against this kind of indentation. I know the codebase is full of them (hello @Jokeswar) but let's not add more. Here's a nicer way:

```python
                return render(
                    request,
                    'interface/upload.html',
                    {'form': form, 'delta_t': delta_t},
                )
```